### PR TITLE
JSON protocol updating

### PR DESCRIPTION
**Pull request by denismakogon**
this patch adds HTTP query string into payload (see more TODOs in code)
 adds one more test to verify query

### CODE REVIEW

File: api/agent/protocol/json.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ func writeString(err error, dst io.Writer, str string) error {
 	return err
 }
 
+// TODO(xxx): headers, query parameters, body - what else should we add to func's payload?
```
Comment (hibooboo2) [inline api/agent/protocol/json.go]: Request url request method shouldnt they be included?

File: api/agent/protocol/json_test.go [hunk 2]
```diff
@@ -38,10 +40,17 @@ func TestJSONProtocolDumpJSONRequestWithData(t *testing.T) {
 		Host: "localhost:8080",
 	}
 	var buf bytes.Buffer
-	rDataBefore := RequestData{A: "a"}
-	json.NewEncoder(&buf).Encode(rDataBefore)
+
+	if data != nil {
+		_ = json.NewEncoder(&buf).Encode(data)
```
Comment (hibooboo2) [inline api/agent/protocol/json_test.go]: What happens if this is not able to encode / errors? shouldnt you handle?

File: api/agent/protocol/json.go (hunk 1 above)
Comment (denismakogon) [reply api/agent/protocol/json.go]: They are part of env: FN_METHOD, FN_REQUEST_URL, etc.

Comment (hibooboo2) [reply api/agent/protocol/json.go]: K

File: api/agent/protocol/json_test.go (hunk 2 above)
Comment (denismakogon) [reply api/agent/protocol/json_test.go]: Well, data being encoded here is test data, it should not fail.

File: api/agent/protocol/json.go (hunk 1 above)
Comment (hibooboo2) [reply api/agent/protocol/json.go]: Just made sense to have at top level to me but it's the either way so :).

File: api/agent/protocol/json_test.go (hunk 2 above)
Comment (hibooboo2) [reply api/agent/protocol/json_test.go]: Right. Gotcha 👍

File: api/agent/protocol/json.go (hunk 1 above)
Comment (denismakogon) [reply api/agent/protocol/json.go]: Other thing is auth: user/password from URL structure

Comment (treeder) [reply api/agent/protocol/json.go]: Request url should be in Json since it can change every request.

Comment (denismakogon) [reply api/agent/protocol/json.go]: @treeder most parts of URL already in env like HTTP method, request URL, which parts of it should also be included?

**Pull request merged**